### ISSUE TITLE
User Preferences-Adding External Accounts

### ISSUE THREAD

**Issue by parvathyvr** [eclipse-che/che#17954]
### Is your enhancement related to a problem? Please describe.
The users should be able to set up the various external accounts they want to link to Che using User Preferences. This currently includes:
- GitHub account
- External Image Registries
- SSH Key Pairs
- Git credentials
- Maven settings
- Generic secret file
- Generic secret environment variable

### Describe the solution you'd like
The user can add various external account types under 'User Preferences'.

**Step 1**
Could be similar to oc login display token where we expose the cmd for them to copy/paste into a terminal to add a secret. We should also explain how to query for created secrets
https://github.com/eclipse/che/issues/20469

**Step 2**
Would add a YAML editor where a user could add content and apply. A quick way that we could provide additional help/value would be to pre-fill the form with a skeleton secret structure which includes the following:

```yaml
apiVersion: v1
kind: Secret
metadata:
  generatedName: che-user-secret-
  labels:
    app.kubernetes.io/part-of: che.eclipse.org
    app.kubernetes.io/component: workspace-secret
```

At this stage we should also provide a table(?) of created secrets which the user would be able to view/edit/delete. This should update after a new secret has been added through the above form. 

Creation / deletion of secrets should provide feedback inc. timeout error message.

**Step 3**

would start to add basic validation as described below:

We should check for these labels and show an error message if they are not there (error message should include these so they can be easily added):

```yaml
    app.kubernetes.io/part-of: che.eclipse.org
    app.kubernetes.io/component: workspace-secret 
```

And we should check if the annotation “che.eclipse.org/mount-as” and we should show an error message if it’s not there and if the value is neither “file” or “env”

```
che.eclipse.org/mount-as: file | env
```

If it’s a file then we should also check that the annotation che.eclipse.org/mount-path is there 

```
che.eclipse.org/mount-path: ...
```

If it’s an env we should check that at least one annotation with the suffix env-name is there:

```
che.eclipse.org/*env-name:...
```

**Step 4**

Would add the ability to select the type of secret that is being created (i.e. drop down menu of type, or + button then menu. Which asks for specific information they would already know about or know how to retrieve, and automatically create the full secret for them:

### Additional context
An exploratory design concept is attached below

![Che-User preferences Ext accounts Design concept](https://user-images.githubusercontent.com/33114242/94163548-9c167d80-fea5-11ea-9d10-dc5bbfa65b49.png)

**Comment by l0rd** [comment]
@parvathyvr I would update the table like that:

| Type | Description | Details | Action |
| --- | --- | --- | :--: |
| GitHub | Signed in to **GitHub** as **l0rd** | [Token](url) | : |
| Container Image Registry | Signed in to **quay.io** as **mloriedo** | [Credentials](url) | : |
| SSH Key Pair | `SHA256:Up6KjbnEV4Hgfo75YM393QdQsK3Z0aTNBz0DoirrW+c val@klar` | [Key pair](url) | : |
| SSH Configuration | User SSH config file | [File details](url) | : |
| Git Configuration | **mario.loriedo@gmail.com** | [Credentials](url) | : |
| Git Credentials | username: **mloriedo** | [Credentials](url) | : |
| Maven Settings | settings.xml file | [File details](url) | : |
| Generic Secret File | File mounted at ~/.my_private_secret | [File details](url) | : |
| Generic Secret Env Vars | File mounted at ~/.ssh/config | [File details](url) | : |

@benoitf @davidfestal @ericwill @skabashnyuk @sleshchenko @azatsarynnyy any feedback?

**Comment by sympatheticmoose** [comment]
@l0rd I'm wondering about "Add Workspaces Secret". Which whilst accurate, offers little abstraction to the user from the mechanism we're using in kubernetes - which they shouldn't _have_ to understand, as isn't necessarily a clear 'action' they would want to take to achieve (for example) connecting their Github account.

**Comment by l0rd** [comment]
@sympatheticmoose agreed. I haven't found anything better though. "Add External Account" doesn't work for tools settings that doesn't involve an external service (settings or git settings for example)

**Comment by themr0c** [comment]
Is it possible to reconsider the priority to P1?

In the documentation currently have several procedures for the end-user containing some steps for "adding {orch-name} Secret containing __<foo>__".
It implies a major context change for the user: switching from the Dashboard, to  CLI  tools to convert the secret to base64, and to configure Kubernetes/OpenShift.
This change of context is a major drawback for the end-user.

For the naming: what about simply: "Secrets".

For reference, the list of all pages containing a mention to provisioning a Secret:

### End-user Guide

For all procedures, the user needs to expose secrets in a workspace.

* https://www.eclipse.org/che/docs/che-7/end-user-guide/mounting-a-secret-as-a-file-or-an-environment-variable-into-a-workspace-container/
* https://www.eclipse.org/che/docs/che-7/end-user-guide/authenticating-on-scm-server-with-a-personal-access-token/
* https://www.eclipse.org/che/docs/che-7/end-user-guide/provisioning-jetbrains-activation-code-for-offline-use/
* https://www.eclipse.org/che/docs/che-7/end-user-guide/using-maven-artifact-repositories/
* https://www.eclipse.org/che/docs/che-7/end-user-guide/using-gradle-artifact-repositories/
* https://www.eclipse.org/che/docs/che-7/end-user-guide/using-python-artifact-repositories/
* https://www.eclipse.org/che/docs/che-7/end-user-guide/using-go-artifact-repositories/
* https://www.eclipse.org/che/docs/che-7/end-user-guide/using-nuget-artifact-repositories/
* https://www.eclipse.org/che/docs/che-7/end-user-guide/using-npm-artifact-repositories/

### Installation guide

Mostly unavoidable steps during installation, except for the last in the list, related to the actions in the  End-user Guide.

* https://www.eclipse.org/che/docs/che-7/installation-guide/installing-che-on-kubespray/
* https://www.eclipse.org/che/docs/che-7/installation-guide/installing-che-on-aws/
* https://www.eclipse.org/che/docs/che-7/installation-guide/installing-che-on-google-cloud-platform/
* https://www.eclipse.org/che/docs/che-7/installation-guide/installing-che-on-microsoft-azure/
* https://www.eclipse.org/che/docs/che-7/installation-guide/configuring-che-hostname/
* https://www.eclipse.org/che/docs/che-7/installation-guide/deploying-che-with-support-for-git-repositories-with-self-signed-certificates/
* https://www.eclipse.org/che/docs/che-7/installation-guide/importing-untrusted-tls-certificates/
* https://www.eclipse.org/che/docs/che-7/installation-guide/mounting-a-secret-as-a-file-or-an-environment-variable-into-a-container/

### Administration guide:


* https://www.eclipse.org/che/docs/che-7/administration-guide/external-database-setup/
* https://www.eclipse.org/che/docs/che-7/administration-guide/configuring-authorization/
* https://www.eclipse.org/che/docs/che-7/administration-guide/configuring-openshift-oauth/
